includes/admin/helper/class-wc-helper.php

```diff
@@ -1007,13 +1007,18 @@ private static function _get_subscriptions_from_product_id( $product_id, $single
 	}
 
 	/**
-	 * Obtain a list of locally installed Woo extensions.
+	 * Obtain a list of data about locally installed Woo extensions.
 	 */
 	public static function get_local_woo_plugins() {
 		if ( ! function_exists( 'get_plugins' ) ) {
 			require_once ABSPATH . 'wp-admin/includes/plugin.php';
 		}
 
+		$woo_plugins = get_transient( 'wc_helper_installed_woo_plugins' );
```

Comment:
This will probably cause problems with the Helper.

We have no reliable way of determining when a new plugin is installed, or when one is deleted, and the Helper relies on this data always being accurate, so caching it, even for just a day, is going to cause confusion when users purchase a new extension, but the Helper doesn't see it for up to 24h.

The Helper will only call this method in the plugin (extension/subscription) management context, much like WordPress core. By context I mean the extension management screen in Woo core, the plugin list in wp-admin, as well as the plugin update checks in Cron. Caching this data for those routines has no real benefit in my opinion.

The problem we're trying to solve is that, what has previously been called in plugin management only, is now being called on many other non-plugin management screens in wp-admin, so caching sounds like a good solution. If we do cache it however, let's try and do this "internal" to the marketplace suggestions module to avoid any side effects on the Helper code.

Reply:
Now fixed – I've reverted these changes, now we're using `active_plugins` option. 8dbda1b9c962a93be7d17ffa8ac55c2639f641e1